open-apparel-registry/open-apparel-registry
# Implement Source model - Step 3

## Overview

The third step of a three step process to add a Source model between facility lists and list items, allowing for the submission of individual facilities.

Each step in the process is intended to be released separately and maintain backward compatibility.

This is a "cleanup" step that removes fields made redundant by the Source model.

Connects #817
Depends on #856
Depends on #857

## Notes

This is the third of three pull requests to be deployed separately.

This PR is targeted at `jcw/source-model-2` because it depends on it. 

NOTE that this includes destructive migration. Going back to another develop-based branch will require a `resetdb`.

## Testing Instructions


* Checkout `develop` and run `./scripts/resetdb` and `./scripts/manage liststosources`
* Checkout this branch. Run `./scripts/manage migrate`
* Exercise every list/item/facility related feature
  * Log in as `c2@example.com`
  * Search for `regina miracle intimate` and view the details of the matching facility. Verify that the "Other locations," "Also known as," "Other addresses," and "Contributors" sections are populated.
  * Click the "Claim this facility" link and verify that you can successfully submit the claim.
  * In a new private browser window
    * Log in as `c1@example.com`.
    * Browse http://localhost:6543/dashboard/claims/1 and approve the claim
  * Back in the original browser window, refresh the "Regina" facility details page and click "Update facility details"
    * Enter a few field values and save the form
  * Copy the OAR ID
  * Browse http://localhost:8081/api/docs/#!/facilities/facilities_get_facility_history and submit the OAR ID. Verify that the history contains `CREATE`, `ASSOCIATE`, `CLAIM`, and `CLAIM_UPDATE`	events.
  * Browse http://localhost:6543/contribute and upload [other-location-test.csv.zip](https://github.com/open-apparel-registry/open-apparel-registry/files/3692205/other-location-test.csv.zip)
  * Fully process the list. Verify there are no processing errors.
    * `./scripts/manage batch_process --list-id 16 --action parse`
    * `./scripts/manage batch_process --list-id 16 --action geocode`
    * `./scripts/manage batch_process --list-id 16 --action match`
  * Browse http://localhost:6543/lists/16
    * Reject one of the pending matches, accept the other. Verify that the process completes without error.
  * Back in the private browser window (superuser account) browse http://localhost:8081/admin/reports/. 
    * Run all the reports and verify that they don't return errors. NOTE several of the reports will not contain any results when run against the development database.



## Checklist
